src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -77,6 +101,27 @@ module _ {a b} {A : Set a} {B : Set b} (f : A → B) where
   map⁺ (swap x y p)  = swap _ _ (map⁺ p)
   map⁺ (trans p₁ p₂) = trans (map⁺ p₁) (map⁺ p₂)
 
+  -- permutations preserve 'being a mapped list'
+  map⁻ : ∀ {xs ys} → map f xs ↭ ys → ∃ λ ys′ → ys ≡ map f ys′ × xs ↭ ys′
```

Comment:
Given our discussion #1133 I think this  shouldn't be called `map⁻` as the precondition is not of the form `map f xs ↭ map f ys`. It's actually much more similar to the form of the other `inv`  lemmas you've added. How  about `↭-map-inv`?

Reply:
Should that just be `map-inv` to be more consistent with the other properties?

Reply:
I'm not entirely sure that `map-inv` is better than `map⁻`. Perhaps the other properties should be `empty⁻` and `singleton⁻`? @gallais?

Reply:
> Should that just be map-inv to be more consistent with the other properties?

You've prepended `↭-` to the other inverse lemmas (e.g. `↭-singleton-inv`) so I would for this one as well?

> I'm not entirely sure that map-inv is better than map⁻

It's not so much that it's better per se. It's just that the superscript ⁻ is reserved for lemmas of a very particular form that this lemma does not obey. I'm happy to discuss alternatives for `↭-map-inv`, it's just that I'm very against `map⁻` and `empty⁻` and `singleton⁻`. 

For example  the name `singleton⁻` should be reserved for the type `[ x ] ↭ [ y ] → x ≡ y`. As far as I can think `map⁻` and `singleton⁻` have no appropriate type.

Reply:
Ok, thanks!